src/DotVVM.Framework/Routing/RouteHelper.cs

```diff
@@ -35,5 +36,54 @@ public static void AutoDiscoverRoutes(this DotvvmRouteTable table, IRoutingStrat
                 table.Add(route.RouteName, route);
             }
         }
+
+        public static void AssertConfigurationIsValid(this DotvvmConfiguration config)
+        {
+            var invalidRoutes = new List<DotvvmConfigurationAssertResult<RouteBase>>();
+            var invalidControls = new List<DotvvmConfigurationAssertResult<DotvvmControlConfiguration>>();
+            var loader = new AggregateMarkupFileLoader();
+            foreach (var route in config.RouteTable.Where(s => !string.IsNullOrWhiteSpace(s.VirtualPath)))
+            {
+                if (string.IsNullOrWhiteSpace(route.RouteName))
+                {
+                    invalidRoutes.Add(new DotvvmConfigurationAssertResult<RouteBase>(route, DotvvmConfigurationAssertReason.MissingRouteName));
+                }
+
+                var content = loader.GetMarkup(config, route.VirtualPath);
+                if (content == null)
+                {
+                    invalidRoutes.Add(new DotvvmConfigurationAssertResult<RouteBase>(route, DotvvmConfigurationAssertReason.MissingFile));
+                }
+            }
+            foreach (var control in config.Markup.Controls.Where(s => !string.IsNullOrWhiteSpace(s.Src)))
+            {
+                if (string.IsNullOrWhiteSpace(control.TagPrefix))
+                {
+                    invalidControls.Add(new DotvvmConfigurationAssertResult<DotvvmControlConfiguration>(control, DotvvmConfigurationAssertReason.MissingControlTagPrefix));
+                    break;
+                }
+
+                if (string.IsNullOrWhiteSpace(control.TagName) && !string.IsNullOrWhiteSpace(control.Src))
```

Comment:
Is there the `!string.IsNullOrWhiteSpace(control.Src)` check really needed? It is filtered with `Where` in `foreach` statement

Reply:
The Where condition should not be there. Because empty tagname and existing src means that you missed to set tagname.